File: [프로그래머스] 연습문제_롤케이크 자르기.py
from collections import Counter 
def solution(topping):
    answer = 0
    toppingDic = Counter(topping)
    toppingSet = set()
    for i in topping:
        toppingDic[i] -= 1
        toppingSet.add(i)
        if toppingDic[i] == 0:
            toppingDic.pop(i)
        if len(toppingDic) == len(toppingSet):
               answer += 1
    return answer

File: [프로그래머스] 연습문제_롤케이크 자르기.js
function solution(topping) {
    var answer = 0;
    let toppingDic = {}
    topping.forEach((i) => {
        i in toppingDic === false ? toppingDic[i] = 1 : toppingDic[i] += 1;
    })
    let toppingSet = new Set();
    let dicLen = Object.keys(toppingDic).length
    for(let i of topping){
        toppingDic[i] -= 1;
        toppingSet.add(i);
        if(toppingDic[i] === 0){
            dicLen -= 1;
        }
        if(dicLen === toppingSet.size){
            answer += 1;
        }
    }
    return answer;
}